actix-service/src/and_then.rs

```diff
@@ -261,6 +261,91 @@ where
     }
 }
 
+/// `.and_then_send()` service factory combinator
+pub struct AndThenSendServiceFactory<A, B, Req>
```

Comment:
why not just use Arc in AndThenServiceFactory?

Reply:
It just seems unfortunate to have to downgrade the performance of and_then. I'm still tinkering though.

Reply:
Since ServiceFactory::new_service is O(1) during app lifetime, I think it is reasonable to have more cost there to guarantee `Send`abliity. The output service type can keep using Rc shich is what matters!

Reply:
of course, this would be an extra pointer indirection per-request Rc<Arc<>> :(